mne/time_frequency/spectrum.py

```diff
@@ -1368,6 +1418,84 @@ def average(self, method="mean"):
         return Spectrum(state, **defaults)
 
 
+@fill_doc
+class EpochsSpectrumArray(EpochsSpectrum):
+    """EpochsSpectrum object from numpy array.
+
+    Parameters
+    ----------
+    data : array, shape (n_epochs, n_channels, n_freqs)
+        The channels' power spectral density for each epoch.
+    %(info_not_none)s
+    %(freqs_tfr)s
+    %(method_psd)s
+    events : None | array of int, shape (n_events, 3)
+        The events typically returned by the read_events function.
+        If some events don't match the events of interest as specified
+        by event_id, they will be marked as 'IGNORED' in the drop log.
+        If None (default), all event values are set to 1 and event time-samples
+        are set to range(n_epochs).
+    event_id : int | list of int | dict | None
+        The id of the event to consider. If dict,
+        the keys can later be used to access associated events. Example:
+        dict(auditory=1, visual=3). If int, a dict will be created with
+        the id as string. If a list, all events with the IDs specified
+        in the list are used. If None, all events will be used with
+        and a dict is created with string integer names corresponding
+        to the event id integers.
+    metadata : instance of pandas.DataFrame | None
+        See :class:`mne.Epochs` docstring for details.
+    %(selection)s
+    %(drop_log)s
```

Comment:
wondering what makes the most sense for these.  Would like opinions of other devs (@hoechenberger @larsoner @mscheltienne @agramfort) as well as @alexrockhill... Off the top of my head:

- `events` I can see why folks might want to pass this in but is there any reason it needs to be a "proper" events array (N x 3)?  Seems like only the last column is going to matter? it *can* be set after-the-fact so we *could* omit it from the constructor, but my gut is that this will be frequently used so maybe best to leave it here?
- `event_id` again I can see why folks might want this (for `__getitem__`)... it can be set after-the-fact and probably doesn't hurt to allow that; not sure about *forcing* it to be after-the-fact if we're going to have `events` as a param though. Thoughts?
- `metadata` can be set after instantiation so I'm inclined to leave it out of the constructor
- `selection` seem like this shouldn't be needed if you're working with `np.array` data. Is there a use case you're thinking of where users would want to pass anything besides `selection=np.arange(data.shape[0])`? If not we can just set that internally
- `drop_log` similar to `selection`; is there a use case you're thinking of where users would want to pass anything besides `drop_log=tuple(tuple() for _ in range(data.shape[0]))`? If not we can just set that internally

If we do end up keeping `events` and `event_id` as-is, the docstring should be deduped with `EpochsArray` (where it seems you copy-pasted these entries from)

Reply:
+1 on simplifying the constructor, as inferred, this was copy-pasted with the thought that it should just be similar to `EpochsArray` and without much more thought than that. I think `events` and `event_id` are helpful in the constructor so people don't lose expected functionality.